obal/data/module_utils/obal.py

```diff
@@ -15,10 +15,16 @@ def specfile_macro_lookup(specfile, macro_str):
         'dist',
         '--undefine',
         'foremandist',
-        '--srpm',
-        specfile
+        '--srpm'
     ]
-    return subprocess.check_output(cmd, universal_newlines=True)
+
+    if scl is not None:
+        cmd.append('--define="scl %s"' % scl)
+
+    cmd.append(specfile)
+    name = subprocess.check_output(' '.join(cmd), universal_newlines=True, shell=True)
```

Comment:
Please don't join. There should be no reason to and it breaks handling of directories/files with spaces.

Reply:
If you can make this work without joining I will avoid it, but I *cannot* get this to work with how rpmbuild/rpmspec has macros defined using `--define`.

Reply:
I think you should try `cmd.append('--define') ; cmd.append("scl %s"' % scl)`